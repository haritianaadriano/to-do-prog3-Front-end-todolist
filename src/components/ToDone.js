export default function ToDone({done}){
    return(
        <>
            <h3>Done</h3>
            <ul
                data-testid="todone-id"
            >
                {done.map((todo, index) => (
                <li key={index}>{todo}</li>
                ))}
            </ul>
        </>
    )
}