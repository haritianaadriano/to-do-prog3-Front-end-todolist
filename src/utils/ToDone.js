export default function ToDone(){
    return(
        <>
            
        </>
    )
}