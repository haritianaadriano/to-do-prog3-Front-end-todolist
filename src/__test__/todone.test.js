import {render, cleanup, screen} from "@testing-library/react";
import {create} from 'react-test-renderer';
import ToDone from "../components/ToDone";

describe("To Done test", () => {
    const mock = ["mock-1", "mock-2"];

    it("ToDone renders correctly", () => {
        render(<ToDone done={mock}/>);
        const todoneElement = screen.getByTestId("todone-id");

        expect(todoneElement).toBeInTheDocument();
    })

    it("[snapshot] ToDone renders correctly", () => {
        const renderer = create(<ToDone done={mock}/>);
        const json = renderer.toJSON();

        expect(json).toMatchSnapshot();
    })

    it("ToDone renders correctly the list", () => {
        render(<ToDone done={mock}/>);
        const todoneElement = screen.getByTestId("todone-id");
        const [element1, element2] = mock;

        expect(todoneElement.children).toHaveLength(mock.length);
        expect(todoneElement).toHaveTextContent(element1);
        expect(todoneElement).toHaveTextContent(element2);
    })
})