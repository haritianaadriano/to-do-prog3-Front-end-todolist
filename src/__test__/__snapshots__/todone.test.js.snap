// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`To Done test [snapshot] ToDone renders correctly 1`] = `
Array [
  <h3>
    Done
  </h3>,
  <ul
    data-testid="todone-id"
  >
    <li>
      mock-1
    </li>
    <li>
      mock-2
    </li>
  </ul>,
]
`;
